frontend/src/components/common/StateHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import type { ApolloError } from "@apollo/client";
+
+import ErrorState from "./info/ErrorState";
+import LoadingState from "./info/LoadingState";
+
+interface StateHandlerProps {
+  loading: boolean;
+  error?: ApolloError;
+  isEmpty?: boolean;
+  emptyState?: React.ReactNode;
+  children: React.ReactNode | React.ReactNode[];
+  fullPage?: boolean;
+}
+
+const StateHandler = ({
```

Comment:
`StateHandler` is a very confusing name out of context. Could we change this to something clearer? Some bad ideas (please improve these more! Ask chatgpt :P):
 - DataStateFallbackWrapper
 - DataLoadingGate
 - PageDataGate

Reply:
any of DataStatusRenderer, QueryStateHandler, ConditionalRenderer?

Reply:
`QueryStateHandler` seems better, yeah